File: 0003.无重复字符的最长子串/0003-无重复字符的最长子串.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ret = 0
        i = 0
        d = {}
        for j in range(len(s)):
            if s[j] in d:
                i = max(i, d[s[j]] +1)
            ret = max(ret, j-i+1)
            d[s[j]] = j
        return ret

File: 0003.无重复字符的最长子串/0003-无重复字符的最长子串.rs
use std::collections::HashSet;
impl Solution {
    pub fn length_of_longest_substring(s: String) -> i32 {
        let mut ret = 0;
        let mut l = 0;
        let mut set = HashSet::new();
        let chars: Vec<_>  = s.chars().collect();
        for c in chars.iter() {
            while set.contains(c) {
                set.remove(&chars[l]);
                l += 1;
            }
            set.insert(c);
            ret = ret.max(set.len());
        }
        
        ret as i32
    }
}